lib/home/library/video_library.dart

```diff
@@ -118,110 +110,43 @@ class _VideoLibraryState extends State<VideoLibrary> {
   Future<void> fetchDocuments() async {
     final QuerySnapshot snapshot = await Firestore.streams.get();
     setState(() {
-      documents = snapshot.docs;
-    });
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    if (documents.isEmpty) {
-      fetchDocuments();
-      return const Center(
-        child: CircularProgressIndicator(),
-      );
-    } else {
-      if (allVideos.isEmpty) {
-        for (document in documents) {
-          allVideos.add(VideoCard(
+      videos = allVideos = [
+        for (final document in snapshot.docs)
+          VideoCard(
             id: document.id,
             title: document['title'],
             category: document['category'],
             director: document['director'],
             path: document['storage_path'],
             timestamp: document['date'],
-          ));
-        }
-
-        videos = allVideos;
-      }
-
-      return DefaultTabController(
-          length: 2,
-          child: Scaffold(
-            appBar: AppBar(
-              title: const Text('Video Library'),
-            ),
-            body: Column(
-              children: <Widget>[
-                Container(
-                  margin: const EdgeInsets.fromLTRB(16, 16, 16, 0),
-                  child: TextField(
-                    controller: controller,
-                    decoration: InputDecoration(
-                      prefixIcon: const Icon(Icons.search),
-                      hintText: 'Search here',
-                      border: OutlineInputBorder(
-                        borderRadius: BorderRadius.circular(8),
-                      ),
-                    ),
-                    onChanged: searchVideo,
-                  ),
-                ),
-                DropdownButton<String>(
-                  value: selectedCategory,
-                  onChanged: (newValue) {
-                    setState(() {
-                      selectedCategory = newValue!;
-                      filterVideos();
-                    });
-                  },
-                  items: _buildCategoryDropdownItems(),
-                ),
-                Expanded(
-                  child: ListView.builder(
-                      itemCount: videos.length,
-                      itemBuilder: (context, index) {
-                        final video = videos[index];
-                        return video;
-                      }),
-                ),
-              ],
-            ),
-          ));
-    }
+          ),
+      ];
+    });
   }
 
-  List<DropdownMenuItem<String>> _buildCategoryDropdownItems() {
+  List<String> get categories {
     final categories = {
       'All',
       for (final video in allVideos) video.category,
     };
-    final items = [
-      for (final category in categories.toList()..sort())
-        DropdownMenuItem<String>(value: category, child: Text(category)),
-    ];
-    return items;
+    return categories.toList()..sort();
   }
```

Comment:
The original function was something [I'd suggested previously](https://github.com/theheartcenter-inc/thc/pull/45#discussion_r1588497099). Hopefully soon I'll stop changing my mind :)